src/gemdat/transitions.py

```diff
@@ -405,3 +406,197 @@ def _calculate_occupancy(atom_sites: np.ndarray) -> dict[int, int]:
+        List of the peaks that correspond to high probability regions
+    percolate_x : bool
+        If True, consider paths that percolate along the x dimension
+    percolate_y : bool
+        If True, consider paths that percolate along the y dimension
+    percolate_z : bool
+        If True, consider paths that percolate along the z dimension
+
+    Returns
+    -------
+    total_energy_cost: float
+        Total energy cost of the best percolating path
+    best_starting_point: tuple
+        Coordinates of the starting site where the best percolating path was found
+    best_perc_path: list
+        List of coordinates of the path
+    best_perc_path_energy: list
+        Energy along the path
+    """
+    X_real, Y_real, Z_real = F.shape
+
+    # Find percolation using virtual images along the required dimensions
+    if not (percolate_x + percolate_y + percolate_z):
+        print('Warning: percolation is not defined')
+        return None, None, None
+    if percolate_x:
+        extended_F = np.empty((2 * F.shape[0], F.shape[1], F.shape[2]),
+                              dtype=F.dtype)
+        extended_F[:X_real, :, :] = F
+        extended_F[X_real:, :, :] = F
+        F = extended_F
+    if percolate_y:
+        extended_F = np.empty((F.shape[0], 2 * F.shape[1], F.shape[2]),
+                              dtype=F.dtype)
+        extended_F[:, :Y_real, :] = F
+        extended_F[:, Y_real:, :] = F
+        F = extended_F
+    if percolate_z:
+        extended_F = np.empty((F.shape[0], F.shape[1], 2 * F.shape[2]),
+                              dtype=F.dtype)
+        extended_F[:, :, :Z_real] = F
+        extended_F[:, :, Z_real:] = F
+        F = extended_F
+    X, Y, Z = F.shape
+
+    # Get F on a graph
+    F_graph = free_energy_graph(F, max_energy_threshold=1e7, diagonal=True)
+
+    # Find the lowest cost path that percolates along the x dimension
+    total_energy_cost = float('inf')
+    best_starting_point = None
+    best_perc_path = []
+
+    for starting_point in peaks:
+
+        # Get the end point which is a periodic image of the peak
+        end_point = (starting_point[0] + X_real * percolate_x,
+                     starting_point[1] + Y_real * percolate_y,
+                     starting_point[2] + Z_real * percolate_z)
+
+        # Find the shortest percolating path through this peak
+        try:
+            path, path_energy = optimal_path(
+                F_graph,
+                tuple(starting_point),
+                tuple(end_point),
+            )
+        except nx.NetworkXNoPath:
+            continue
+
+        # Calculate the path cost
+        path_cost = np.sum(path_energy)
+
+        if path_cost < total_energy_cost:
+            total_energy_cost = path_cost
+            best_starting_point = starting_point
+            best_perc_path = path
+            best_perc_path_energy = path_energy
+
+    return total_energy_cost, best_starting_point, best_perc_path, best_perc_path_energy
```

Comment:
Try to avoid returning a tuple. It's often easier to return an object, like a namedtuple or dataclass and access these variables via an attribute.

Reply:
Created a path class